sdk/storage/Azure.Storage.Common/src/Shared/UserDelegationKeyProperties.cs

```diff
@@ -0,0 +1,71 @@
+﻿// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+
+using System;
+using System.Globalization;
+using System.Net;
+using System.Text;
+
+namespace Azure.Storage.Sas
+{
+    /// <summary>
+    /// Encapsulates the shared properties used by both
+    /// BlobSasQueryParameters and DataLakeSasQueryParameters.
+    /// </summary>
+    internal struct UserDelegationKeyProperties
```

Comment:
Make this a `class` (too many members) and make everything properties instead of fields.